src/App.tsx

```diff
@@ -1,14 +1,75 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+/* eslint-disable react-hooks/rules-of-hooks */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  ChangeEvent, useEffect, useMemo, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Status } from './types/Status';
+import { ErrorMessages } from './types/ErrorMessages';
+
+const USER_ID = 11217;
+
+const filterTodos = (todos: Todo[], selectedStatus: Status) => {
+  let filteredTodos = [...todos];
+
+  switch (selectedStatus) {
+    case Status.Completed:
+      filteredTodos = filteredTodos.filter((todo: Todo) => todo.completed);
+      break;
+    case Status.Active:
+      filteredTodos = filteredTodos.filter((todo: Todo) => !todo.completed);
+      break;
+    default:
+      break;
+  }
 
-const USER_ID = 0;
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.All);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState<ErrorMessages | null>(null);
+  const timerRef = useRef<number | null>(null);
+  const [isEditable, setIsEditable] = useState(false);
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todos, selectedStatus),
+    [selectedStatus, todos],
+  );
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos);
+  }, [todos, isEditable]);
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const showError = () => {
+    setError(ErrorMessages.Add);
+
+    // Clear the existing timer, if any
+    if (timerRef.current) {
+      clearTimeout(timerRef.current);
+    }
+
+    // Set a new timer to auto-hide the error message after 3 seconds
+    timerRef.current = window.setTimeout(() => {
+      setError(null);
+    }, 3000);
```

Comment:
no need to use ref. try to set timeout for error disappearance in `ErrorMessages`component on mount